theano/gpuarray/sort.py

```diff
@@ -0,0 +1,308 @@
+from __future__ import absolute_import, print_function, division
+import os
+from string import Template
+
+import numpy as np
+from theano import Apply
+from theano.tensor import as_tensor_variable
+from theano.tensor.sort import TopKOp
+
+from .basic_ops import (GpuKernelBase, Kernel, infer_context_name,
+                        as_gpuarray_variable)
+from .opt import register_opt, op_lifter, register_opt2
+from .type import GpuArrayType
+
+try:
+    import pygpu
+    import pygpu.gpuarray as ga
+except ImportError as e:
+    # To make sure theano is importable
+    pass
+
+# TODO sort / argsort
+
+# TODO add runtime opt, if k==1, use max/min reduce
+#      also if k is axis size, just copy input tensor
```

Comment:
Do you know if in real use case, will k be constant most of the time? If so, we could use a Theano optimizer that when k is constant in the graph, do the optimization at the graph level. That would be much simpler then at run time.